script/zip-comp-logs

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python3
+"""
+A script to zip the logs produced by a series of competition matches.
+"""
+
+import re
+import shutil
+import argparse
+import datetime
+import contextlib
+from typing import Union, Iterator
+from pathlib import Path
+from zipfile import ZipFile
+
+
+def zip_folder(folder: Path, args: argparse.Namespace) -> str:
+    tla = folder.name
+    zip_name = f'team-{tla}'
+
+    if args.suffix:
+        zip_name += f'-{args.suffix}'
+    if args.include_date:
+        zip_name += f'-{datetime.date.today()}'
+
+    shutil.make_archive(
+        f'{args.output_dir / zip_name}',
+        'zip',
+        root_dir=folder,
+        base_dir='.',
+    )
+    return f'{zip_name}.zip'
+
+
+def is_team_logs_folder(folder: Path) -> bool:
+    # Options for detecting which folders are logs:
+    # 1. everthing except recordings, matches
+    # 2. folders w/ matching zip file names
+    # 3. regex of folder name
+    # 4. regex of folder contents names
+
+    # Using option 3
+    # folder names are three uppercase letters and an optional number
+    if re.match(r'[A-Z]{3}[0-9]*$', folder.name):
```

Comment:
I think this regex wants to be slightly different so we can be more sure of its match

```suggestion
    if re.match(r'^[A-Z]{3}[0-9]?$', folder.name):
```
Also, you can just return the output of this directly rather than needing the `if`

Reply:
`re.match` matches from the start of the string so the `'^'` is implicit, if we want to make that explicit we should probably use `re.search`

Reply:
It feels a bit odd to me to have an explicit `$` but not a matching `^`. I think it would be reasonable to use `re.match` here, even with explicit `^`.